core/src/main/java/com/adobe/aem/commons/assetshare/components/actions/share/impl/EmailShareServiceImpl.java

```diff
@@ -189,7 +189,7 @@ private final String getAssetLinkListHtml(final Config config, final String[] as
                 if (isAuthor()) {
                     url = externalizer.authorLink(config.getResourceResolver(), url);
                 } else {
-                    url = externalizer.publishLink(config.getResourceResolver(), url);
+                    url = externalizer.externalLink(config.getResourceResolver(), cfg.externalizerDomain(), url);
```

Comment:
This is still going to be limited to a single key, which can be a problem when multiple domains are in play.

What do you think about creating a new `AssetShareExternalizer` service that has an `externalLink` function that runs the logic as we see here.  That would allow anyone then to create a new implementation of that service with a higher rank to override the logic, allowing the custom implementation to externalize the link however it sees fit.

Reply:
I think that's a great idea - my issue with it is that this implementation allows for a simple config file to live in whatever codebase leveraging it (or a manual config on the server) whereas if we want people to implement a service then we are requiring a dependency in someone's codebase on this code. I'm not sure if that is the recommended approach unless this code (asset share itself) is already planned to be embedded in the customer's code.

Reply:
we could implement both - abstracting this code to a service does not mean it cant still use your fancy config :D

Reply:
I think we could implement a simple request host to externalizer domain mapping service that would allow us to still just add a config in the customized code instead of adding a dependency and extending a class.

Reply:
After chatting w/Ryan, agreed to stick with the simpler solution for now.